src/test/scala/com/fulcrumgenomics/commons/reflect/ReflectionUtilTest.scala

```diff
@@ -61,6 +61,42 @@ object ReflectionUtilTest {
 object UsingCompanion { def apply(s: String): UsingCompanion = new UsingCompanion(s.toInt)}
 case class UsingCompanion(x: Int)
 
+/** Sealed-trait hierarchy with no companion */
+sealed trait NoCompanion
+case object ANoCompanion extends NoCompanion
+case object BNoCompanion extends NoCompanion
+case object CNoCompanion extends NoCompanion
+
+/** Sealed-trait hierarchy with a companion but neither a `values` nor a `findValues` method */
+sealed trait EmptyCompanion
+case object AEmptyCompanion extends EmptyCompanion
+case object BEmptyCompanion extends EmptyCompanion
+case object CEmptyCompanion extends EmptyCompanion
+object NoCompanion
+
+/** Sealed-trait hierarchy where `values` returns the values in the order the case objects they were declared */
+sealed trait CompanionWithValues
+case object ACompanionWithValues extends CompanionWithValues
+case object BCompanionWithValues extends CompanionWithValues
+case object CCompanionWithValues extends CompanionWithValues
+object CompanionWithValues {
+  def values: Seq[CompanionWithValues] = Seq(ACompanionWithValues, BCompanionWithValues, CCompanionWithValues)
+}
+
+/** Sealed-trait hierarchy where `findValues` returns the values in a different order than the case objects they were declared */
+sealed trait CompanionWithFindValues
+case object ACompanionWithFindValues extends CompanionWithFindValues
+case object BCompanionWithFindValues extends CompanionWithFindValues
+case object CCompanionWithFindValues extends CompanionWithFindValues
+object CompanionWithFindValues {
+  def findValues: Seq[CompanionWithFindValues] = Seq(BCompanionWithFindValues, ACompanionWithFindValues, CCompanionWithFindValues)
+}
+
```

Comment:
Should we also test cases where the case objects are declared inside the companion object?  That's the usual pattern for enumeratum enums, so it would be good to make sure it works for those.

Reply:
Good idea, done.